eclipse-theia/theia
# Ensure `Disposable.NULL` always returns a new `Disposable`

#### What it does


Because of the way `DisposableCollections` handle disposal of their members, it is unsafe to add `Disposable.NULL` to any disposable collection: doing so risks disposal of one collection marking another collection as disposed inappropriately. This PR adds a test showing the risk of `Disposable.NULL` and replaces `Disposable.NULL` throughout the framework with a new method `Disposable.createNull`, with a test to demonstrate that disposables created in that way are safe for use in `DisposableCollections`.

#### How to test


1. Assert that the tests in [`disposable.spec.ts`](https://github.com/eclipse-theia/theia/pull/11053/files#diff-c94e7d7c6a8101807f20839ba79577f5565e362b980f004ff2d5f04bb0ff3484) pass (and look correct)
2. Check that changes other than `disposable.ts` and `dispospable.spec.ts` just replace `Disposable.NULL` with `Disposable.createNull`
3. Check that no references to `Disposable.NULL` remain in the framework.
4. Check that the application behaves as expected.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
